File: cole/day01_contains-duplicate/contains-duplicate.cpp
// beats 97.60%

#include <unordered_set>

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> m;
        m.reserve(nums.size());
        for (auto it = nums.begin(); it != nums.end() && m.insert(*it).second; ++it);
        return m.size() != nums.size();
    }
};

File: Jaxon/Day1 - Contains Duplicate/ContainsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        seen_vals = []
        for val in range(len(nums)):
            counter = 0
            for num in nums:
                if num == nums[val]:
                    counter += 1
            if counter > 1:
                return True
        return False 

            

            